src/frontend/src/lib/services/address.services.ts

```diff
@@ -147,26 +159,51 @@ const loadIdbTokenAddress = async <T extends Address>({
 			`Error encountered while searching for locally stored ${tokenId.description} public address in the browser.`
 		);
 
-		return { success: false };
+		return { success: false, err: tokenId };
 	}
 
 	return { success: true };
 };
 
-export const loadIdbBtcAddressMainnet = async (): Promise<ResultSuccess> =>
+const loadIdbBtcAddressMainnet = async (): Promise<ResultSuccess<TokenId>> =>
 	loadIdbTokenAddress<BtcAddress>({
 		tokenId: BTC_MAINNET_TOKEN_ID,
 		getIdbAddress: getIdbBtcAddressMainnet,
 		updateIdbAddressLastUsage: updateIdbBtcAddressMainnetLastUsage
 	});
 
-export const loadIdbEthAddress = async (): Promise<ResultSuccess> =>
+const loadIdbEthAddress = async (): Promise<ResultSuccess<TokenId>> =>
 	loadIdbTokenAddress<EthAddress>({
 		tokenId: ETHEREUM_TOKEN_ID,
 		getIdbAddress: getIdbEthAddress,
 		updateIdbAddressLastUsage: updateIdbEthAddressLastUsage
 	});
 
+export const loadIdbAddresses = async (): Promise<ResultSuccess<TokenId[]>> => {
+	const results = await Promise.all([
+		NETWORK_BITCOIN_ENABLED ? loadIdbBtcAddressMainnet() : { success: true, err: null },
+		loadIdbEthAddress()
+	]);
+
+	let success = true;
+	const err: TokenId[] = [];
+
+	results.map(({ success: s, err: e }) => {
```

Comment:
Can you resolve this in a functional non imperative way?

Reply:
right! better!